content/articles/building-and-deploying-a-language-detection-system-using-flask/index.md

```diff
@@ -0,0 +1,322 @@
+	- [Modeling](#modeling)
+	- [Model evaluation](#model-evaluation)
+	- [Inference](#inference)
+- [Deploying the model using Flask](#deploying-the-model-using-flask)
+
+### Prerequisites
+To follow along with this article, you should have the following:
+- Basic knowledge of Machine Learning
+- Python and [Jupyter Notebook](https://jupyter.org/)]( installed on your machine
+- Know the basics of [Flask](https://flask.palletsprojects.com/en/2.0.x/tutorial/)
+
+### Building the language detection model
+It would be best if you started by setting up a development environment.
+
+#### Setting up the development environment
+The first step will be to create a project folder and create a virtual environment to handle the project dependencies.
+
+Run the following commands to create the project's folder:
+
+```bash
+mkdir langdetect
+cd langdetect
+```
+
+In the newly created folder `langdetect`, create a virtual environment and activate it using the following commands:
+
+```bash
+py -m venv .env #create a virtual environment named .env
+cd .env/Scripts/activate # activate the environment
+```
+
+Next,  create a new file, `languagedetection.ipynb` in the `langdetect` folder. Then, open the file using Jupyter Notebook.
+
+#### Importing libraries
+Before starting to model, we need to import the libraries listed below:
+
+```Python
+import numpy as np
+import pandas as pd
+import seaborn as sns
+import re          #(regular expression)
+import matplotlib.pyplot as plt
+import warnings
+warnings.simplefilter("ignore")
+```
+
+Next, read the dataset using the following code snippet:
+
+```python
+df= pd.read_csv("Language Detection.csv")
+df.head()
+```
+
+We will then check the number of languages in the dataset using the following code snippet:
+
+```Python 
+df["Language"].value_counts()
+```
+
+Our dataset has 17 languages: English, French, Spanish, Portuguese, Italian, Russian, Swedish, Malayalam, Dutch, Arabic, Turkish,   German, Tamil, Danish, Kannada, Greek, and Hindi. 
+
+Next, proceed to preprocess our data.
+
+#### Data preprocessing
+We will differentiate the dataset features from the labels as shown below:
+
+```Python
+X = df["Text"]
+y = df["Language"]
+```
+
+Next, we will encode the label (**Language**) using the `LabelEncoder()` method as shown below:
+
+```python
+from sklearn.preprocessing import LabelEncoder
+le = LabelEncoder()
+y = le.fit_transform(y)
+```
+In the next step, we need to preprocess our feature (**Text**) as shown below:
+
+```Python
+text_list = []
+
+# iterating through all the text
+for text in X:         
+    text = re.sub(r'[!@#$(),n"%^*?:;~`0-9]', ' ', text)      # removes all the symbols and numbers
+    text = re.sub(r'[[]]', ' ', text)   
+    text = text.lower()          # converts all the text to lower case
+    text_list.append(text)       # appends the text to the text_list
+```
+Then, we encode our feature(**Text**) through a *Bag of Words* model using the `CountVectorizer()` method as shown below:
+
```

Comment:
The steps carried out here are fine and the snippets are great :+1: However, it would really be helpful if you explained why:
- You differentiate the dataset features from the labels
- What is encoding and how is it useful. What exactly happens when you encode?
- What pre-processing is and why we need to conduct it.
- What is Bag of Words?